hummingbot/connector/exchange/bybit/bybit_api_order_book_data_source.py

```diff
@@ -0,0 +1,315 @@
+                        else:
+                            self.logger().error("There was an error subscribing to the topic "
+                                                f"{json_message['request']['args']} ({json_message['ret_msg']})")
+                    else:
+                        topic = json_message["topic"]
+                        topic = ".".join(topic.split(".")[:-1])
+                        self._messages_queues.get(topic).put_nowait(json_message)
+
+            except asyncio.CancelledError:
+                raise
+            except Exception as ex:
+                self.logger().network(
+                    f"Unexpected error with WebSocket connection ({ex})",
+                    exc_info=True,
+                    app_warning_msg="Unexpected error with WebSocket connection. Retrying in 30 seconds. "
+                                    "Check network connection."
+                )
+                if ws_adaptor:
+                    await ws_adaptor.close()
+                await asyncio.sleep(30.0)
+
+    async def listen_for_trades(self, ev_loop: asyncio.BaseEventLoop, output: asyncio.Queue):
+        """
+        Process trade events received through the websocket channel
+        """
+        symbol_map = await self.trading_pair_symbol_map(self._domain)
+
+        while True:
+            try:
+                trade_message = await self._messages_queues[CONSTANTS.WS_TRADES_TOPIC].get()
+                for trade_entry in trade_message["data"]:
+                    trade_msg: OrderBookMessage = BybitOrderBook.trade_message_from_exchange(
+                        msg=trade_entry,
+                        timestamp=trade_entry["trade_time_ms"] * 1e-3,
+                        metadata={"trading_pair": symbol_map[trade_entry["symbol"]]})
+                    output.put_nowait(trade_msg)
+            except asyncio.CancelledError:
+                raise
+            except Exception as ex:
+                self.logger().error(f"Unexpected error ({ex})", exc_info=True)
+
+    async def listen_for_order_book_diffs(self, ev_loop: asyncio.BaseEventLoop, output: asyncio.Queue):
+        """
+        Listen for order book diffs events received through the websocket channel
+        """
+        symbol_map = await self.trading_pair_symbol_map(self._domain)
+
+        while True:
+            try:
+                order_book_message = await self._messages_queues[CONSTANTS.WS_ORDER_BOOK_EVENTS_TOPIC].get()
+
+                symbol = order_book_message["topic"].split(".")[-1]
+                trading_pair = symbol_map[symbol]
+                event_type = order_book_message["type"]
+
+                if event_type == "snapshot":
+                    snapshot_msg: OrderBookMessage = BybitOrderBook.snapshot_message_from_exchange(
+                        msg=order_book_message,
+                        timestamp=order_book_message["timestamp_e6"] * 1e-6,
```

Comment:
We discussed this with Daniel. We turn the timestamp into seconds (instead of microseconds) to allow anyone working with the message to create a datetime from the timestamp if required. It is a matter of consistency really: in every place in the client when we refer to something as a timestamp we are referring to a number of seconds. Also the logic in datetime to create an instance from a timestamp fails if the number specified is not an amount of senconds.